File: kyu6/BackspacesInString/Kata.cs
public class Kata
{
  public static string CleanString(string s)
  {
    string res = "";
    for (int i = 0 ;  i < s.Length; i++){
        if (s[i]=='#')res = res.Substring(0, res.Length-1>-1?res.Length-1:0);
        else res = res+s[i];
    }
    return res;
  }
}

File: kyu6/BackspacesInString/kata.rb
def clean_string(string)
  r = ""
  x = string.length - 1
  for i in 0..x do
    if string[i].eql?('#')
      r = r[0..-2]
    else
      r = r + string[i]
    end
  end
  r
end